enterprise/backend/src/metabase_enterprise/workspaces/api.clj

```diff
@@ -141,6 +143,34 @@
     {:transforms (for [transform transforms]
                    (assoc transform :workspace (get workspaces-by-id (tid->wid (:id transform)))))}))
 
+(api.macros/defendpoint :post "/:id/promote"
+  :- [:map
+      [:promoted [:sequential [:map [:id ms/PositiveInt] [:name :string]]]]
+      [:errors {:optional true} [:sequential [:map [:id ms/PositiveInt] [:name :string] [:error :string]]]]
+      [:workspace [:map [:id ms/PositiveInt] [:name :string]]]
+      [:archived-at :any]]
+  "Promote workspace transforms back to main Metabase and archive the workspace.
+
+  This will:
+  1. Update original transforms with workspace versions
+  2. Re-execute transforms in the original schema
+  3. Archive the workspace and clean up isolated resources
+
+  Returns a report of promoted transforms and any errors."
+  [{:keys [id]} :- [:map [:id ms/PositiveInt]]
```

Comment:
This route doesn't have `transform` in the URL, so I don't think we should couple it to transforms - even "just for now".

Can you make this rather take an "entities map", like create does? `{:transforms [1,2,3,4]}`

Reply:
I also wonder whether we want to use this API for "promote everything", and what that call pattern would look like then.

Reply:
OK I was too hasty - this is taking a workspace ID, not a transform.

We want the ability to promote partial sets (in particular single entities) at a time.